test-suite/shared/src/test/scala/org/scalajs/testsuite/scalalib/RangesTest.scala

```diff
@@ -35,4 +40,19 @@ class RangesTest {
 
     assertEquals(List(0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0), range.toList)
   }
+
+  @Test def NumericRange_overflow_issue_2407(): Unit = {
+    val nr = NumericRange(Int.MinValue, Int.MaxValue, 1 << 23)
+    assertEquals(Int.MinValue, nr.sum)
+  }
+
+  @Test def Range_foreach_issue_2409(): Unit = {
+    assumeFalse(executingInJVM && scalaVersion.startsWith("2.10"))
```

Comment:
Ideally, this should be `"2.10."` with a trailing `.`, so that `"2.100.0"` is not matched.